pkg/cluster/px/px.go

```diff
@@ -769,22 +769,26 @@ func (ops *pxClusterOps) isScaleDownOfSharedAppsRequired(isMajorVerUpgrade bool,
 	return opts.SharedAppsScaleDown == SharedAppsScaleDownOn
 }
 
-// isMajorVersionUpgrade checks if this is a 1.2 to 1.3/1.4 upgradee
-func (ops *pxClusterOps) isMajorVersionUpgrade(spec *apiv1alpha1.Cluster) (bool, error) {
+// doesUpgradeNeedAppDrain checks if target is 1.3.3 or upgrade is from 1.2 to 1.3/1.4
+func (ops *pxClusterOps) doesUpgradeNeedAppDrain(spec *apiv1alpha1.Cluster) (bool, error) {
 	currentVersionDublin, err := ops.isAnyNodeRunningVersionWithPrefix("1.2")
 	if err != nil {
 		return false, err
 	}
 
-	newMajorMinor, err := parseMajorMinorVersion(spec.Spec.OCIMonTag)
+	newVersion, err := parsePXVersion(spec.Spec.OCIMonTag)
 	if err != nil {
 		return false, err
 	}
 
-	logrus.Infof("Is any node running dublin version: %v. new version (major.minor): %s",
-		currentVersionDublin, newMajorMinor)
+	logrus.Infof("Is any node running dublin version: %v. new version: %s", currentVersionDublin, newVersion)
 
-	return currentVersionDublin && (newMajorMinor == "1.3" || newMajorMinor == "1.4"), nil
+	if strings.HasPrefix(newVersion, "1.3.3") {
+		// 1.3.3 has a change that requires a reboot even if starting version is not dublin
+		return true, nil
+	}
+
+	return currentVersionDublin && (strings.HasPrefix(newVersion, "1.3") || strings.HasPrefix(newVersion, "1.4")), nil
```

Comment:
Please double-check the boolean logic here.
- earlier we checked explicitly for version "1.3.3", and returned TRUE if matched
- here we are checking if version is DUBLIN, _as well_ as have version-strings "1.3.*" or "1.4.*"

.. should't this be?
```go
// NOTE: dublin || 1.3.* || 1.4.*
return currentVersionDublin || strings.HasPrefix(newVersion, "1.3") || strings.HasPrefix(newVersion, "1.4"), nil
```

Reply:
The check is correct. It looks for 2 conditions

1. New version starts with 1.3.3. In this case, regardless of what the start version is, return that drain is required.
2. New version starts with 1.3 or 1.4 and start version is dublin.